src/Cli/dotnet/commands/dotnet-tool/restore/ToolRestoreCommand.cs

```diff
@@ -61,6 +63,15 @@ public ToolRestoreCommand(
             _configFilePath = result.GetValue(ToolRestoreCommandParser.ConfigOption);
             _sources = result.GetValue(ToolRestoreCommandParser.AddSourceOption);
             _verbosity = result.GetValue(ToolRestoreCommandParser.VerbosityOption);
+            if (!result.HasOption(ToolRestoreCommandParser.VerbosityOption) && result.GetValue(ToolCommandRestorePassThroughOptions.InteractiveRestoreOption))
+            {
+                _verbosity = VerbosityOptions.minimal;
+            }
+
+            _restoreActionConfig = new RestoreActionConfig(DisableParallel: result.GetValue(ToolCommandRestorePassThroughOptions.DisableParallelOption),
+                NoCache: result.GetValue(ToolCommandRestorePassThroughOptions.NoCacheOption) || result.GetValue(ToolCommandRestorePassThroughOptions.NoHttpCacheOption),
+                IgnoreFailedSources: result.GetValue(ToolCommandRestorePassThroughOptions.IgnoreFailedSourcesOption),
+                Interactive: result.GetValue(ToolCommandRestorePassThroughOptions.InteractiveRestoreOption));
```

Comment:
This is great! I have two broad comments:

* I feel like I've seen this 'verbosity needs to be downstream of some other configuration' problem in a few different places in the codebase - is it something that you think we could detect and extract to prevent playing whack-a-mole?
* the set of parameters to the RestoreActionConfig seem very 'regular' (in the sense of common/not-changing) - is there a way we could enforce that
  * commands that use RestoreActionConfig have the required options applied to them and
  * something in the way those commands are processed/executed automatically configures a RestoreActionConfig? Again focusing on 'How can we be consistent by design' here instead of playing whack-a-mole. Maybe a middleware or extension method on ParseResult or something?